src/Raven.Client/DocumentStoreBase.cs

```diff
@@ -34,6 +34,35 @@ protected DocumentStoreBase()
             Subscriptions = new DocumentSubscriptions(this);
         }
 
+        public virtual void OnBeforeStore(Documents.InMemoryDocumentSessionOperations session, string id, object entityinstance, IDictionary<string, string> metadata)
+        {
+            BeforeStoreEvent?.Invoke(session, id, entityinstance, metadata);
+        }
+
+        public virtual void OnAfterStore(Documents.InMemoryDocumentSessionOperations session, string id, object entityinstance, IDictionary<string, string> metadata)
+        {
+            AfterStoreEvent?.Invoke(session, id, entityinstance, metadata);
+        }
+
+        public bool BeforeStoreEmpty => (BeforeStoreEvent == null);
```

Comment:
This is pretty much only relevant if you need to do some amount of work before calling the event, so it isn't worth it if there are no subscribers